examples/csrf.js

```diff
@@ -8,35 +8,36 @@ var jade = require('../'),
     Compiler = jade.Compiler,
     nodes = jade.nodes;
 
-var options = {
-    compiler: CSRF
-  , locals: {
-    csrf: 'WAHOOOOOO'
-  }
-};
-
-jade.renderFile(__dirname + '/csrf.jade', options, function(err, html){
-    if (err) throw err;
-    console.log(html);
-});
-
-function CSRF(node, options) {
+var CSRF = function CSRF(node, options) {
   Compiler.call(this, node, options);
 }
 
-CSRF.prototype.__proto__ = Compiler.prototype;
+CSRF.prototype = Object.create(Compiler.prototype);
 
 CSRF.prototype.visitTag = function(node){
   var parent = Compiler.prototype.visitTag;
-  switch (node.name) {
-    case 'form':
-      if ("'post'" == node.getAttribute('method')) {
+  if (node.name === 'form') {
+    for (var i = 0; i < node.attrs.length; i++) {
+      var attr = node.attrs[i];
+      if (attr && attr.name === 'method' && attr.val === "'post'") {
         var tok = new nodes.Tag('input');
-        tok.setAttribute('type', '"hidden"');
-        tok.setAttribute('name', '"csrf"');
-        tok.setAttribute('value', 'csrf');
+        tok.setAttribute('type', '"hidden"', true);
+        tok.setAttribute('name', '"csrf"', true);
+        tok.setAttribute('value', 'csrf', false);
```

Comment:
This API should be considered deprecated.  I'm planning to completely overhall the AST as part of the 2.0.0 rewrite.  We should probably just remove this example.

I would like to support plugins properly afterwards though, and as part of that I'd like to consider the AST format part of the public API.

It might be worth doing something to add a deprecation notice to `.setAttribute`, but we should be careful to make sure it doesn't get output when jade uses the method internally.